retrieval/target.go

```diff
@@ -359,9 +364,15 @@ func (t *Target) scrape(sampleAppender storage.SampleAppender) (err error) {
 	}()
 
 	for samples := range t.ingestedSamples {
-		for _, s := range samples {
-			s.Metric.MergeFromLabelSet(baseLabels, clientmodel.ExporterLabelPrefix)
-			sampleAppender.Append(s)
+		if t.proxy {
+			for _, s := range samples {
+				sampleAppender.Append(s)
```

Comment:
It's not. The explicitly empty label is gone after ingestion, you federate, new instance attached. This adds up to 3 already. Even if you have a boolean flag, for every future change you can introduce the next flag even though it's internal.

The number of `mode`s are entirely in our control - no one said anything about adding a new mode for every exporter.